src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala

```diff
@@ -0,0 +1,118 @@
+        (17d, false, true),
+        (17.98d, false, true),
+        (18d, false, true),
+        (20d, false, false),
+        (22d, true, false),
+        (22.02d, true, false),
+        (23d, true, false),
+      )
+
+      testCases.foreach { case (innerTemperature, isTooHigh, isTooLow) =>
+        val innerTemp = Temperature(innerTemperature, Celsius)
+        val isHigher = thermalHouseTest.isInnerTemperatureTooHigh(innerTemp)
+        val isLower = thermalHouseTest.isInnerTemperatureTooLow(innerTemp)
+
+        isHigher shouldBe isTooHigh
+        isLower shouldBe isTooLow
+      }
+    }
+
+    "Calculation of thermal energy change and new inner temperature is performed correctly" in {
+      val thermalHouseTest = thermalHouse(18, 22)
+      val innerTemperature = Temperature(20, Celsius)
+
+      val thermalEnergyGain =
+        thermalHouseTest.calcThermalEnergyGain(Kilowatts(100), Seconds(3600))
+      val thermalEnergyLoss = thermalHouseTest.calcThermalEnergyLoss(
+        innerTemperature,
+        Temperature(10, Celsius),
+        Seconds(3600),
+      )
+      val thermalEnergyChange = thermalHouseTest.calcThermalEnergyChange(
+        thermalEnergyGain,
+        thermalEnergyLoss,
+      )
+      val innerTemperatureChange =
+        thermalHouseTest.calcInnerTemperatureChange(thermalEnergyChange)
+      val newInnerTemperature = thermalHouseTest.calcNewInnerTemperature(
+        innerTemperature,
+        innerTemperatureChange,
+      )
+
+      thermalEnergyGain should approximate(KilowattHours(100))
+      thermalEnergyLoss should approximate(KilowattHours(10))
+      thermalEnergyChange should approximate(KilowattHours(90))
+      innerTemperatureChange should approximate(Kelvin(9.0))
+      newInnerTemperature should approximate(Celsius(29.0))
+    }
+
+    "Comprising function to calculate new inner temperature works as expected" in {
+      val thermalHouseTest = thermalHouse(18, 22)
+      val thermalPower = Kilowatts(100)
+      val duration = Seconds(3600)
+      val currentInnerTemperature = Temperature(20, Celsius)
+      val ambientTemperature = Temperature(10, Celsius)
+
+      val newInnerTemperature = thermalHouseTest.newInnerTemperature(
+        thermalPower,
+        duration,
+        currentInnerTemperature,
+        ambientTemperature,
+      )
+
+      newInnerTemperature should approximate(Temperature(29, Celsius))
+    }
+
+    "Check build method" in {
+
+      val thermalHouseTest = thermalHouse(18, 22)
+      val thermalHouseInput = defaultThermalHouse
+
+      thermalHouseTest.id shouldBe thermalHouseInput.getId
+      thermalHouseTest.operatorInput shouldBe thermalHouseInput.getOperator
+      thermalHouseTest.operationTime shouldBe thermalHouseInput.getOperationTime
+      thermalHouseTest.bus shouldBe null
+      thermalHouseTest.ethLosses.toWattsPerKelvin shouldBe 1000.0
+      (thermalHouseTest.ethCapa * Temperature(
+        1,
+        Celsius,
+      )).toKilowattHours shouldBe 2741.5
```

Comment:
Be careful here, original test is with 1 Kelvin which is different to what happens if one uses 1 Celsius.

Reply:
Yeah Thanks. :)